dummyserver/testcase.py

```diff
@@ -160,21 +161,25 @@ class HTTPDummyServerTestCase:
     server: ClassVar[httpserver.HTTPServer]
     port: ClassVar[int]
     server_thread: ClassVar[threading.Thread]
+    _stack: ClassVar[contextlib.ExitStack]
 
     @classmethod
     def _start_server(cls) -> None:
-        cls.io_loop = ioloop.IOLoop.current()
-        app = web.Application([(r".*", TestingApp)])
-        cls.server, cls.port = run_tornado_app(
-            app, cls.io_loop, cls.certs, cls.scheme, cls.host
-        )
-        cls.server_thread = run_loop_in_thread(cls.io_loop)
+        with contextlib.ExitStack() as stack:
+            io_loop = stack.enter_context(run_loop_in_thread())
```

Comment:
ExitStack is used so we can use `with ` for the `run_loop_in_thread()` context manager and close it if anything fails during setup then `cls._stack = stack.pop_all()` is used to close the context manager later

Reply:
So if we had a pytest fixture here instead of setup_class/teardown_class, could we use `run_loop_in_thread` as a normal context manager? If yes, can we please do that?

Reply:
yes, can I leave that for another PR?